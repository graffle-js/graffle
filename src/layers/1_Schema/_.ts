export { type RootTypeName } from './core/helpers.js'
export * from './core/Named/__.js'
export * as Directives from './Directives.js'
export * from './Hybrid/__.js'
export * from './Hybrid/types/Enum.js'
export * from './Hybrid/types/Scalar/__.js'
export * from './Input/__.js'
export * from './Input/Args.js'
export * from './Input/types/InputObject.js'
export * from './Output/__.js'
export * from './Output/types/__typename.js'
export * from './Output/types/Field.js'
export * from './Output/types/Interface.js'
export { type Object$2 } from './Output/types/Object.js'
export * from './Output/types/Union.js'
