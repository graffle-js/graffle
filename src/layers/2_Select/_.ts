export * from './$parseSelection.js'
export * as Arguments from './arguments.js'
export * from './Directive/__.js'
export * as Document from './document.js'
export * from './Indicator/__.js'
export * as InlineFragment from './InlineFragment.js'
export * as SelectAlias from './SelectAlias.js'
export * as SelectionSet from './selectionSet.js'
export * as SelectScalarsWildcard from './SelectScalarsWildcard.js'
// todo resolve this export
export { type Bases } from './selectionSet.js'
