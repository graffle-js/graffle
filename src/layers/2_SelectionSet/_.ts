export { toGraphQLDocument } from './graphqlMapper/nodes/Document.js'
export * from './nodes/__.js'
export type { Directive } from './nodes/Directive/__.js'
export * as Document from './nodes/document.js'
export * from './nodes/Indicator/__.js'
export * as On from './nodes/InlineFragment.js'
export * from './nodes/selectionSet.js'
