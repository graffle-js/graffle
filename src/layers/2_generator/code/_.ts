import { createModuleGenerator } from '../createCodeGenerator.js'
import { moduleNameClient } from './Client.js'
import { moduleNameError } from './Error.js'
import { moduleNameSelect } from './Select.js'

export const { generate: generate_, moduleName: moduleName_ } = createModuleGenerator(
  `_`,
  ({ code }) => {
    code.push(
      `// We import the global module for good measure although it is not clear it is always needed.`,
      `// It at least helps with Twoslash wherein without this import here Twoslash will not include the global module.`,
      `// In real TypeScript projects it seems the global module is included automatically. But there could be certain tsconfig`,
      `// setups where this still indeed does help.`,
      `import './modules/Global.js'`,
      ``,
      `export { Select } from './modules/${moduleNameSelect}.js'`,
      `export { isError } from './modules/${moduleNameError}.js'`,
      `export { create } from './modules/${moduleNameClient}.js'`,
    )

    return code
  },
)
