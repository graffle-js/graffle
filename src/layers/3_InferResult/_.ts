export * from './Alias.js'
export * from './Field.js'
export * from './Interface.js'
export * from './Object.js'
export * from './root.js'
export * from './ScalarsWildcard.js'
export * from './Union.js'
