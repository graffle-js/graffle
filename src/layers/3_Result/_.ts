export * from './infer/_.js'
