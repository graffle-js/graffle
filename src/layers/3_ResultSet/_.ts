export { decode } from './decode.js'
export * from './infer/_.js'
