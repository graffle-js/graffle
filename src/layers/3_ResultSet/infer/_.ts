export * from './Field.js'
export * from './root.js'
export * from './SelectAlias.js'
