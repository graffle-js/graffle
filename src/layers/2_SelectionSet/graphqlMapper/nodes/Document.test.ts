import { print } from 'graphql'
import { describe, expect, test } from 'vitest'
import { db } from '../../../../../tests/_/schemas/db.js'
import { $Index as schemaIndex } from '../../../../../tests/_/schemas/kitchen-sink/graffle/modules/SchemaRuntime.js'
import type * as SelectionSets from '../../../../../tests/_/schemas/kitchen-sink/graffle/modules/SelectionSets.js'
import { toGraphQLDocument } from '../../_.js'
import { createDocumentNormalizedFromRootTypeSelection } from '../../nodes/document.js'
import type { Context } from '../types.js'

type CasesQuery = [selectionSet: SelectionSets.Query]

type CasesDescriptiveQuery = [description: string, selectionSet: SelectionSets.Query]

const testEachArgs = [
  `Query %s`,
  (...args: CasesQuery | CasesDescriptiveQuery) => {
    const [description, selectionSet] = args.length === 1 ? [undefined, args[0]] : args

    const context: Context = {
      schema: schemaIndex,
      captures: {
        variables: [],
        customScalarOutputs: [],
      },
      // config: {
      //   output: outputConfigDefault,
      //   transport: { type: `memory`, config: { methodMode: `post` } },
      //   name: schemaIndex[`name`],

      //   initialInput: {} as any,
      // },
    }
    const documentNormalized = createDocumentNormalizedFromRootTypeSelection(`Query`, selectionSet as any)
    const graphqlDocument = toGraphQLDocument(context, [], documentNormalized)
    const graphqlDocumentStringFormatted = print(graphqlDocument)
    const beforeAfter = `\n`
      + JSON.stringify(selectionSet, null, 2)
      + `\n--------------\n`
      + graphqlDocumentStringFormatted
      + `\n`
    expect(beforeAfter).toMatchSnapshot(description)
  },
] as const

describe(`inline fragment field group`, () => {
  test.each<CasesQuery>([
    [{ ___: { __typename: true } }],
    [{ ___: [{ __typename: true }, { abcEnum: true }] }],
    [{ interface: { ___: { __typename: true } } }],
    [{ unionFooBar: { ___: { __typename: true } } }],
    [{ ___: { __typename: true, $include: true } }],
  ])(...testEachArgs)
})

describe(`enum`, () => {
  test.each<CasesQuery>([
    [{ result: { $: { case: `Object1` }, __typename: true } }],
  ])(...testEachArgs)
})

describe(`union`, () => {
  test.each<CasesQuery>([
    [{ unionFooBar: { __typename: true } }],
    [{ unionFooBar: { ___on_Bar: { int: true } } }],
    [{ unionFooBar: { ___on_Bar: { $skip: true, int: true } } }],
    // s({ unionFooBar: { __on_Bar: {} } }), // todo should be static type error
  ])(...testEachArgs)
})

describe(`alias`, () => {
  test.each<CasesQuery>([
    [{ id: [`x`, true] }],
    [{ id: [[`x`, true], [`id2`, true]] }],
    [{ id: [`x`, { $skip: true }] }],
    [{ object: [`x`, { $skip: true, id: true }] }],
  ])(...testEachArgs)
})

describe(`args`, () => {
  test.each<CasesQuery>([
    [{ stringWithArgs: { $: { boolean: true, float: 1 } } }],
    [{ stringWithArgs: { $: {} } }],
    // s({ objectWithArgs: { $: { id: `` } } }), // todo should be static error
    // s({ objectWithArgs: { $: {} } }), // todo should be static error
    [{ objectWithArgs: { $: { id: `` }, id: true } }],
    [{ objectWithArgs: { $: {}, id: true } }],
  ])(...testEachArgs)
})

describe(`$include`, () => {
  test.each<CasesQuery>([
    [{ object: { $include: true, id: true } }],
    [{ object: { $include: false, id: true } }],
    [{ object: { $include: undefined, id: true } }],
    [{ object: { $include: { if: true }, id: true } }],
    [{ object: { $include: { if: false }, id: true } }],
    [{ object: { $include: { if: undefined }, id: true } }],
    [{ object: { $include: {}, id: true } }],
  ])(...testEachArgs)
})

describe(`$skip`, () => {
  test.each<CasesQuery>([
    [{ object: { $skip: true, id: true } }],
    [{ object: { $skip: false, id: true } }],
    [{ object: { $skip: undefined, id: true } }],
    [{ object: { $skip: { if: true }, id: true } }],
    [{ object: { $skip: { if: false }, id: true } }],
    [{ object: { $skip: { if: undefined }, id: true } }],
    [{ object: { $skip: {}, id: true } }],
  ])(...testEachArgs)
})

// describe(`$defer`, () => {
//   test.each<[SelectionSets.Query]>([
//     [{ object: { $defer: true, id: true } }],
//     [{ object: { $defer: false, id: true } }],
//     [{ object: { $defer: undefined, id: true } }],
//     [{ object: { $defer: { if: true }, id: true } }],
//     [{ object: { $defer: { if: false }, id: true } }],
//     [{ object: { $defer: { if: undefined }, id: true } }],
//     [{ object: { $defer: {}, id: true } }],
//     [{ object: { $defer: { label: `foobar` }, id: true } }],
//   ])(...testEachArgs)
// })

// describe(`$stream`, () => {
//   test.each<[SelectionSets.Query]>([
//     [{ object: { $stream: true, id: true } }],
//     [{ object: { $stream: false, id: true } }],
//     [{ object: { $stream: undefined, id: true } }],
//     [{ object: { $stream: { if: true }, id: true } }],
//     [{ object: { $stream: { if: false }, id: true } }],
//     [{ object: { $stream: { if: undefined }, id: true } }],
//     [{ object: { $stream: {}, id: true } }],
//     [{ object: { $stream: { label: `foobar` }, id: true } }],
//     [{ object: { $stream: { initialCount: 5 }, id: true } }],
//   ])(...testEachArgs)
// })

describe(`other`, () => {
  test.each<CasesQuery>([
    [{ __typename: true }],
    [{ string: true }],
    // [{ string: 1 }],
    // s({ string: false }), // todo should be static error
    [{ id: true, string: false }],
    // [{ id: true, string: 0 }],
    [{ id: true, string: undefined }],
    [{ object: { id: true } }],
    [{ objectNested: { object: { string: true, id: true, int: false } } }],
    [{ objectNested: { object: { string: true, id: true, int: { $skip: true } } } }],
  ])(...testEachArgs)
})

describe(`args`, () => {
  // dprint-ignore
  describe(`custom scalars`, () => {
    test.each<CasesDescriptiveQuery>([
      [`arg field`, { dateArg: { $: { date: db.date0 } } }],
      [`arg field in non-null`, { dateArgNonNull: { $: { date: db.date0 } } }], 
      [`arg field in list`, { dateArgList: { $: { date: [db.date0, new Date(1)] } } }], 
      [`arg field in list (null)`, { dateArgList: { $: { date: null } } }],
      [`arg field in non-null list (with list)`, { dateArgNonNullList: { $: { date: [db.date0, new Date(1)] } } }],
      [`arg field in non-null list (with null)`, { dateArgNonNullList: { $: { date: [null, db.date0] } } }],
      [`arg field in non-null list non-null`, { dateArgNonNullListNonNull: { $: { date: [db.date0, new Date(1)] } } }],
      [`input object field`, { dateArgInputObject: { $: { input: { idRequired: ``, dateRequired: db.date0, date: new Date(1) } } } }],
      [`nested input object field`, { InputObjectNested: { $: { input: { InputObject: { idRequired: ``, dateRequired: db.date0, date: new Date(1) } } } } }]
    ])(...testEachArgs)
  })
})
