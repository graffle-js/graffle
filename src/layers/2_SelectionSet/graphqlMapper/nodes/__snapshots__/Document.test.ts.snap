// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`alias > Query - { id: [ 'x', { 'undefined': true } ] } 1`] = `
"
{
  "id": [
    "x",
    {
      "$skip": true
    }
  ]
}
--------------
{
  x: id @skip(if: true)
}
"
`;

exports[`alias > Query - { id: [ 'x', true ] } 1`] = `
"
{
  "id": [
    "x",
    true
  ]
}
--------------
{
  x: id
}
"
`;

exports[`alias > Query - { id: [ [ 'x', true ], [ 'id2', true ] ] } 1`] = `
"
{
  "id": [
    [
      "x",
      true
    ],
    [
      "id2",
      true
    ]
  ]
}
--------------
{
  x: id
  id2: id
}
"
`;

exports[`alias > Query - { object: [ 'x', { 'undefined': true, id: true } ] } 1`] = `
"
{
  "object": [
    "x",
    {
      "$skip": true,
      "id": true
    }
  ]
}
--------------
{
  x: object @skip(if: true) {
    id
  }
}
"
`;

exports[`arguments > Query - { objectWithArgs: { '$': { id: '' }, id: true } } 1`] = `
"
{
  "objectWithArgs": {
    "$": {
      "id": ""
    },
    "id": true
  }
}
--------------
{
  objectWithArgs(id: "") {
    id
  }
}
"
`;

exports[`arguments > Query - { objectWithArgs: { '$': {}, id: true } } 1`] = `
"
{
  "objectWithArgs": {
    "$": {},
    "id": true
  }
}
--------------
{
  objectWithArgs {
    id
  }
}
"
`;

exports[`arguments > Query - { stringWithArgs: { '$': { boolean: true, float: 1 } } } 1`] = `
"
{
  "stringWithArgs": {
    "$": {
      "boolean": true,
      "float": 1
    }
  }
}
--------------
{
  stringWithArgs(boolean: true, float: 1)
}
"
`;

exports[`arguments > Query - { stringWithArgs: { '$': {} } } 1`] = `
"
{
  "stringWithArgs": {
    "$": {}
  }
}
--------------
{
  stringWithArgs
}
"
`;

exports[`arguments > custom scalars > Query - arg field > arg field 1`] = `
"
{
  "dateArg": {
    "$": {
      "date": "1970-01-01T00:00:00.000Z"
    }
  }
}
--------------
{
  dateArg(date: "1970-01-01T00:00:00.000Z")
}
"
`;

exports[`arguments > custom scalars > Query - arg field in list (null) > arg field in list (null) 1`] = `
"
{
  "dateArgList": {
    "$": {
      "date": null
    }
  }
}
--------------
{
  dateArgList(date: null)
}
"
`;

exports[`arguments > custom scalars > Query - arg field in list > arg field in list 1`] = `
"
{
  "dateArgList": {
    "$": {
      "date": [
        "1970-01-01T00:00:00.000Z",
        "1970-01-01T00:00:00.001Z"
      ]
    }
  }
}
--------------
{
  dateArgList(date: ["1970-01-01T00:00:00.000Z", "1970-01-01T00:00:00.001Z"])
}
"
`;

exports[`arguments > custom scalars > Query - arg field in non-null > arg field in non-null 1`] = `
"
{
  "dateArgNonNull": {
    "$": {
      "date": "1970-01-01T00:00:00.000Z"
    }
  }
}
--------------
{
  dateArgNonNull(date: "1970-01-01T00:00:00.000Z")
}
"
`;

exports[`arguments > custom scalars > Query - arg field in non-null list (with list) > arg field in non-null list (with list) 1`] = `
"
{
  "dateArgNonNullList": {
    "$": {
      "date": [
        "1970-01-01T00:00:00.000Z",
        "1970-01-01T00:00:00.001Z"
      ]
    }
  }
}
--------------
{
  dateArgNonNullList(
    date: ["1970-01-01T00:00:00.000Z", "1970-01-01T00:00:00.001Z"]
  )
}
"
`;

exports[`arguments > custom scalars > Query - arg field in non-null list (with null) > arg field in non-null list (with null) 1`] = `
"
{
  "dateArgNonNullList": {
    "$": {
      "date": [
        null,
        "1970-01-01T00:00:00.000Z"
      ]
    }
  }
}
--------------
{
  dateArgNonNullList(date: [null, "1970-01-01T00:00:00.000Z"])
}
"
`;

exports[`arguments > custom scalars > Query - arg field in non-null list non-null > arg field in non-null list non-null 1`] = `
"
{
  "dateArgNonNullListNonNull": {
    "$": {
      "date": [
        "1970-01-01T00:00:00.000Z",
        "1970-01-01T00:00:00.001Z"
      ]
    }
  }
}
--------------
{
  dateArgNonNullListNonNull(
    date: ["1970-01-01T00:00:00.000Z", "1970-01-01T00:00:00.001Z"]
  )
}
"
`;

exports[`arguments > custom scalars > Query - input object field > input object field 1`] = `
"
{
  "dateArgInputObject": {
    "$": {
      "input": {
        "idRequired": "",
        "dateRequired": "1970-01-01T00:00:00.000Z",
        "date": "1970-01-01T00:00:00.001Z"
      }
    }
  }
}
--------------
{
  dateArgInputObject(
    input: {idRequired: "", dateRequired: "1970-01-01T00:00:00.000Z", date: "1970-01-01T00:00:00.001Z"}
  )
}
"
`;

exports[`arguments > custom scalars > Query - nested input object field > nested input object field 1`] = `
"
{
  "InputObjectNested": {
    "$": {
      "input": {
        "InputObject": {
          "idRequired": "",
          "dateRequired": "1970-01-01T00:00:00.000Z",
          "date": "1970-01-01T00:00:00.001Z"
        }
      }
    }
  }
}
--------------
{
  InputObjectNested(
    input: {InputObject: {idRequired: "", dateRequired: "1970-01-01T00:00:00.000Z", date: "1970-01-01T00:00:00.001Z"}}
  )
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': { if: false }, id: true } } 1`] = `
"
{
  "object": {
    "$include": {
      "if": false
    },
    "id": true
  }
}
--------------
{
  object @include(if: false) {
    id
  }
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': { if: true }, id: true } } 1`] = `
"
{
  "object": {
    "$include": {
      "if": true
    },
    "id": true
  }
}
--------------
{
  object @include(if: true) {
    id
  }
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': { if: undefined }, id: true } } 1`] = `
"
{
  "object": {
    "$include": {},
    "id": true
  }
}
--------------
{
  object @include(if: true) {
    id
  }
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': {}, id: true } } 1`] = `
"
{
  "object": {
    "$include": {},
    "id": true
  }
}
--------------
{
  object @include(if: true) {
    id
  }
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': false, id: true } } 1`] = `
"
{
  "object": {
    "$include": false,
    "id": true
  }
}
--------------
{
  object @include(if: false) {
    id
  }
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': true, id: true } } 1`] = `
"
{
  "object": {
    "$include": true,
    "id": true
  }
}
--------------
{
  object @include(if: true) {
    id
  }
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': undefined, id: true } } 1`] = `
"
{
  "object": {
    "id": true
  }
}
--------------
{
  object {
    id
  }
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': { if: false }, id: true } } 1`] = `
"
{
  "object": {
    "$skip": {
      "if": false
    },
    "id": true
  }
}
--------------
{
  object @skip(if: false) {
    id
  }
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': { if: true }, id: true } } 1`] = `
"
{
  "object": {
    "$skip": {
      "if": true
    },
    "id": true
  }
}
--------------
{
  object @skip(if: true) {
    id
  }
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': { if: undefined }, id: true } } 1`] = `
"
{
  "object": {
    "$skip": {},
    "id": true
  }
}
--------------
{
  object @skip(if: true) {
    id
  }
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': {}, id: true } } 1`] = `
"
{
  "object": {
    "$skip": {},
    "id": true
  }
}
--------------
{
  object @skip(if: true) {
    id
  }
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': false, id: true } } 1`] = `
"
{
  "object": {
    "$skip": false,
    "id": true
  }
}
--------------
{
  object @skip(if: false) {
    id
  }
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': true, id: true } } 1`] = `
"
{
  "object": {
    "$skip": true,
    "id": true
  }
}
--------------
{
  object @skip(if: true) {
    id
  }
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': undefined, id: true } } 1`] = `
"
{
  "object": {
    "id": true
  }
}
--------------
{
  object {
    id
  }
}
"
`;

exports[`enum > Query - { result: { '$': { case: 'Object1' }, __typename: true } } 1`] = `
"
{
  "result": {
    "$": {
      "case": "Object1"
    },
    "__typename": true
  }
}
--------------
{
  result(case: Object1) {
    __typename
  }
}
"
`;

exports[`inline fragment > field group > Query - directive > directive 1`] = `
"
{
  "___": {
    "__typename": true,
    "$include": true
  }
}
--------------
{
  ... @include(if: true) {
    __typename
  }
}
"
`;

exports[`inline fragment > field group > Query - interface > interface 1`] = `
"
{
  "interface": {
    "___": {
      "__typename": true
    }
  }
}
--------------
{
  interface {
    ... {
      __typename
    }
  }
}
"
`;

exports[`inline fragment > field group > Query - multiple > multiple 1`] = `
"
{
  "___": [
    {
      "__typename": true
    },
    {
      "abcEnum": true
    }
  ]
}
--------------
{
  ... {
    __typename
  }
  ... {
    abcEnum
  }
}
"
`;

exports[`inline fragment > field group > Query - one > one 1`] = `
"
{
  "___": {
    "__typename": true
  }
}
--------------
{
  ... {
    __typename
  }
}
"
`;

exports[`inline fragment > field group > Query - union > union 1`] = `
"
{
  "unionFooBar": {
    "___": {
      "__typename": true
    }
  }
}
--------------
{
  unionFooBar {
    ... {
      __typename
    }
  }
}
"
`;

exports[`inline fragment > type condition union > Query - __typename (no fragment) > __typename (no fragment) 1`] = `
"
{
  "unionFooBar": {
    "__typename": true
  }
}
--------------
{
  unionFooBar {
    __typename
  }
}
"
`;

exports[`inline fragment > type condition union > Query - directive > directive 1`] = `
"
{
  "unionFooBar": {
    "___on_Bar": {
      "$skip": true,
      "int": true
    }
  }
}
--------------
{
  unionFooBar {
    ... on Bar @skip(if: true) {
      int
    }
  }
}
"
`;

exports[`inline fragment > type condition union > Query - scalar > scalar 1`] = `
"
{
  "unionFooBar": {
    "___on_Bar": {
      "int": true
    }
  }
}
--------------
{
  unionFooBar {
    ... on Bar {
      int
    }
  }
}
"
`;

exports[`other > Query - { __typename: true } 1`] = `
"
{
  "__typename": true
}
--------------
{
  __typename
}
"
`;

exports[`other > Query - { id: true, string: false } 1`] = `
"
{
  "id": true,
  "string": false
}
--------------
{
  id
}
"
`;

exports[`other > Query - { id: true, string: undefined } 1`] = `
"
{
  "id": true
}
--------------
{
  id
}
"
`;

exports[`other > Query - { object: { id: true } } 1`] = `
"
{
  "object": {
    "id": true
  }
}
--------------
{
  object {
    id
  }
}
"
`;

exports[`other > Query - { objectNested: { object: { string: true, id: true, int: { 'undefined': true } } } } 1`] = `
"
{
  "objectNested": {
    "object": {
      "string": true,
      "id": true,
      "int": {
        "$skip": true
      }
    }
  }
}
--------------
{
  objectNested {
    object {
      string
      id
      int @skip(if: true)
    }
  }
}
"
`;

exports[`other > Query - { objectNested: { object: { string: true, id: true, int: false } } } 1`] = `
"
{
  "objectNested": {
    "object": {
      "string": true,
      "id": true,
      "int": false
    }
  }
}
--------------
{
  objectNested {
    object {
      string
      id
    }
  }
}
"
`;

exports[`other > Query - { string: true } 1`] = `
"
{
  "string": true
}
--------------
{
  string
}
"
`;
